internal/pkg/storageos/volume_labels.go

```diff
@@ -162,3 +180,77 @@ func (c *Client) EnsureReplicas(ctx context.Context, key client.ObjectKey, desir
 	}
 	return observeErr(nil)
 }
+
+// EnsureFailureMode ensures that the desired failure mode has been applied to
+// the StorageOS volume.
+func (c *Client) EnsureFailureMode(ctx context.Context, key client.ObjectKey, desired string) error {
+	funcName := "ensure_failure_mode"
+	start := time.Now()
+	defer func() {
+		metrics.Latency.Observe(funcName, time.Since(start))
+	}()
+	observeErr := func(e error) error {
```

Comment:
Just an idea. What if we give some name to error in the signature and create a `defer` which checks returned error. If not nil then `increment`. With this helper the rest of the code should able to `return err` like any other regular code, while we have the incrementation too.

Reply:
That's a good idea.  We use the current pattern in many places, so let's do a follow-up PR after `v2.4.0` to change everywhere.